### PR TITLE
feat(sparkJobs): Regex filter support for Cardinality Buster

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

With this PR, `delete-pk-filters` filter values can be regex.

### CODE REVIEW

File: spark-jobs/src/main/scala/filodb/cardbuster/PerShardCardinalityBuster.scala [hunk 1]
```diff
@@ -68,17 +65,21 @@ class PerShardCardinalityBuster(dsSettings: DownsamplerSettings,
...
           val schema = schemas(rawSchemaId)
           val pkPairs = schema.partKeySchema.toStringPairs(pk, UnsafeUtils.arayOffset)
-          val willDelete = deleteFilter.exists(filter => filter.forall(pkPairs.contains))
+          val willDelete = deleteFilter.exists(filter => filter.forall { case (filterKey, filterValRegex) =>
+            pkPairs.exists { case (pkKey, pkVal) =>
+              pkKey == filterKey && filterValRegex.findAllMatchIn(pkVal).nonEmpty
+            }
+          })
           if (willDelete) {
             BusterContext.log.debug(s"Deleting part key ${schema.partKeySchema.stringify(pk)}")
```
Comment (sherali42) [inline spark-jobs/src/main/scala/filodb/cardbuster/PerShardCardinalityBuster.scala:84]: can we just log `pkPairs` here?

**Review by sherali42 — APPROVED** [review]
👍

File: spark-jobs/src/main/scala/filodb/cardbuster/PerShardCardinalityBuster.scala (hunk 1 above)
Comment (vishramachandran) [reply spark-jobs/src/main/scala/filodb/cardbuster/PerShardCardinalityBuster.scala]: done. thx

**Review by tjackpaul — APPROVED** [review]
thanks for the quick fix @vishramachandran 👍

**Pull request merged**